### PR TITLE
Listen up

### PR DESCRIPTION
**Pull request by pzavolinsky**
Adds a hacked together `Listen` implementation.

### CODE REVIEW

**Comment by wolfgang42** [comment]
Oog, I didn't even think about async stuff, I was just going to do `process.stdin.readLine()`. (Obviously that wouldn't have worked, I don't know what I was thinking.) Thanks for figuring this one out for me.

It looks great, the only thing is I'm finding the logic in `computeDependencies()` a little hard to follow with the `&&`. I'd prefer one of two ways:

* Push into the array individually:
    ```javascript
    const deps = []
    if (fs) deps.push(`const $fs = require('fs');`)
    return deps
    ```
*  or, map an object:
    ```javascript
    const deps = {fs, addReadLine}
    return Object.keys(deps).filter(k => deps[k]).map(k => depCode[k])
    ```

Obviously this is a matter of opinion but I find these easier to read.

**Comment by wolfgang42** [comment]
Also, please add `computeDependencies` to `module.exports` so it can be used by external libraries. Actually I think it would be better to have two functions, one to compute the dependencies and another to get the code for them, so that alternate implementations can be provided if rockstar-js is used as a library.

File: src/rockstar.js [hunk 1]
```diff
@@ -82,14 +83,40 @@ function groupBlocks(statements) {
...
 }
 
+function computeDependencies(statements) {
+	const addReadLine = statements.some(s => s.t === 'Listen');
+	const fs = addReadLine;
+	
+	return [
+		fs && `const $fs = require('fs');`,
+		addReadLine && `
+		function readLineSync() {
```
Comment (wolfgang42) [inline src/rockstar.js]: This function should probably be `$readLineSync()` the way `$fs` is, to prevent clashes with names used by the program. (Not that I expect someone to name their Rockstar function `readLineSync` but just in case.)

**Comment by pzavolinsky** [comment]
Done!

**Comment by wolfgang42** [comment]
Looks great, thanks a million!
